# Github repository for
---
# Exometabolite dynamics over stationary phase reveal strain-specific responses to nutrient limitation
---
## By John Chodkowski and Ashley Shade
---
*This work is under review*

## Data
Metabolomics data and transcriptomics data are available at Joint Genome Institute Genome Portal under JGI Proposal ID [502921](https://genome.jgi.doe.gov/portal/Grethaproduction/Grethaproduction.info.html).  

Genomes for B. thailandensis, C. violaceum, and P. syringae are available at IMG (58) 
under accession numbers 637000052, 2724679652, and 2508501074, respectively. An improved annotated draft genome of C. violaceum is available under NCBI Project ID 402426 (Genbank Accession ID: PKBZ00000000). Re-sequencing efforts for B. thailandensis and P. syringae are under NCBI project IDs 402425 and 402424, respectively.

## Abstract
Microbial exponential growth is expected to occur infrequently outside of the laboratory, in the environment. Instead, resource-limited conditions impose non-growth states for microbes. However, non-growth states are uncharacterized for the majority of environmental bacteria, especially in regard to exometabolite production. To investigate exometabolite production in response to nutrient limitation, we compared exometabolites produced over time in stationary phase across three environmental of bacteria: Burkholderia thailandensis E264 (ATCC 700388), Chromobacterium violaceum ATCC 31532, and Pseudomonas syringae pathovar tomato DC3000 (ATCC BAA-871). We grew each strain in monoculture and investigated exometabolite dynamics over time from mid-exponential to stationary phase. We focused on exometabolites that were released into the media and accumulated over 45 hours, including approximately 20 hours of stationary phase. In concert, we analyzed transcripts (RNAseq) to inform interpretation of exometabolite output. We found that a majority of exometabolites released under these conditions were strain-specific. A subset of identified exometabolites were involved in both central and secondary metabolism. Transcript analysis supported that exometabolites were released from intact cells, as various transporters were either upregulated or consistently expressed. Interestingly, we found that all strains released succinate, and that each strain re-routed their metabolic pathways involved in succinate production during stationary phase. Overall, these results show that non-growth states can also be metabolically active and dynamic. Furthermore, they show that environmental bacteria have the capability to transform a resource-limited extracellular environment into a rich chemical milieu. This work has implications for understanding microbial community interactions via exometabolites, and within resource-limited environments. 

## Funding

This material is based upon work supported by the National Science Foundation under Grant No DEB#1749544, by Michigan State University, and by a DOE-JGI Community Science Program award (Proposal ID 502921). The work conducted by the U.S. Department of Energy Joint Genome Institute, a DOE Office of Science User Facility, is supported under Contract No. DE-AC02-05CH11231. J.C. was supported by the Eleanor L. Gilmore Fellowship from the Department of Microbiology and Molecular Genetics. 

## Analysis overview

![imageOfworkflow](https://github.com/ShadeLab/Isolate_ExoMetabolitesDynamicsIn_SP/blob/master/images/MSworkflow.png)


